app/scripts/controllers/backup.test.js

```diff
@@ -0,0 +1,30 @@
+import { strict as assert } from 'assert';
+import sinon from 'sinon';
+import BackupController from './backup';
+
+function getMockController() {
+  return {
+    getSelectedAddress: sinon.stub().returns('0x01'),
+    store: {
+      getState: sinon.stub().returns({}),
+    },
+  };
+}
+
+describe('BackupController', function () {
+  const getBackupController = () => {
+    return new BackupController({
+      preferencesController: getMockController(),
+      addressBookController: getMockController(),
+      trackMetaMetricsEvent: sinon.stub(),
+    });
+  };
+
+  describe('constructor', function () {
+    it('should setup correctly', function () {
+      const backupController = getBackupController();
+      const selectedAddress = backupController.preferencesController.getSelectedAddress();
+      assert.equal(selectedAddress, '0x01');
+    });
+  });
```

Comment:
We should add `restoreUserData` and `backupUserData` tests to ensure that we get expected outcomes in each case, since those are the bulk of the work with this PR.

Reply:
The e2e suggestion was made to ensure we have some coverage, but is not a replacement for a unit test.